core/templates/domain/classroom/classroom-backend-api.service.spec.ts

```diff
@@ -93,4 +106,123 @@ describe('Classroom backend API service', function() {
       expect(failHandler).not.toHaveBeenCalled();
     })
   );
+
+  it('should use the fail handler for requests that cannot be processed',
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
+
+      classroomBackendApiService.fetchClassroomData('0').then(
+        successHandler, failHandler);
+
+      let req = httpTestingController.expectOne('/classroom_data_handler/0');
+      expect(req.request.method).toEqual('GET');
+      req.flush('Invalid request', {
+        status: 400,
+        statusText: 'Invalid request'
+      });
+
+      flushMicrotasks();
+
+      expect(successHandler).not.toHaveBeenCalled();
+      expect(failHandler).toHaveBeenCalled();
+    })
+  );
+
+  it('should work with successCallback set to a function',
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success', successCallback);
+      let failHandler = jasmine.createSpy('fail', errorCallback);
```

Comment:
I think you don't need the successCallback and errorCallback here and elsewhere. Can you please check if is it still works just by doing:
```suggestion
      let successHandler = jasmine.createSpy('success');
      let failHandler = jasmine.createSpy('fail');
```